src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2,48 +2,2433 @@
+        verify();
+
+        // expectedOutput: "2"
+        comp = CreateCompilation(new[] { source1, source2 + main });
+        outers = comp.GetMembers("Outer");
+        cs = outers.Select(o => ((NamedTypeSymbol)o).GetMember("C")).ToArray();
+        tree = comp.SyntaxTrees[1];
+        expectedSymbol = cs[1];
+        verify();
+
+        void verify()
+        {
             comp.VerifyDiagnostics();
+            Assert.Equal(2, cs.Length);
+            Assert.Equal(comp.SyntaxTrees[0], cs[0].DeclaringSyntaxReferences.Single().SyntaxTree);
+            Assert.Equal(comp.SyntaxTrees[1], cs[1].DeclaringSyntaxReferences.Single().SyntaxTree);
+
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: true);
+            var cReference = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().Last();
+            var info = model.GetTypeInfo(cReference);
+            Assert.Equal(expectedSymbol.GetPublicSymbol(), info.Type);
         }
     }
+
+    [Fact]
+    public void SignatureUsage_01()
+    {
+        var source = """
+            file class C
+            {
+            }
+
+            class D
+            {
+                public void M1(C c) { } // 1
+                private void M2(C c) { } // 2
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (7,17): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public void M1(C c) { } // 1
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M1").WithArguments("C", "D").WithLocation(7, 17),
+            // (8,18): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     private void M2(C c) { } // 2
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M2").WithArguments("C", "D").WithLocation(8, 18));
+    }
+
+    [Fact]
+    public void SignatureUsage_02()
+    {
+        var source = """
+            file class C
+            {
+            }
+
+            class D
+            {
+                public C M1() => new C(); // 1
+                private C M2() => new C(); // 2
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (7,14): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public C M1() => new C(); // 1
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M1").WithArguments("C", "D").WithLocation(7, 14),
+            // (8,15): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     private C M2() => new C(); // 2
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M2").WithArguments("C", "D").WithLocation(8, 15));
+    }
+
+    // PROTOTYPE(ft): report error on indexer parameters
```

Comment:
>report error on indexer parameters [](http://example.com/codeflow?start=22&length=34)

It looks like the test below covers this case, or perhaps I'm missing something.

Reply:
Looks like this was meant to remind myself to add a test, and I didn't delete this after adding the test. Same with the following line.